zc_common/remote_resource/request.py

```diff
@@ -80,19 +80,30 @@ def add_items_from_data(self):
         map(lambda x: self.append(RemoteResourceWrapper(x)), self.data)
 
 
-def get_route_from_fk(resource_type, pk=None):
-    """Gets a fully qualified URL for a given resource_type, pk"""
-    routes = requests.get(zc_settings.GATEWAY_ROOT_PATH).json()
-
-    for route in routes.iterkeys():
-        if 'resource_type' in routes[route] and routes[route]['resource_type'] == resource_type:
-            if isinstance(pk, (list, set)):
-                expanded = '{}?filter[id__in]={}'.format(expand(route, {}), ','.join([str(x) for x in pk]))
-            else:
-                expanded = expand(route, {'id': pk})
-            return '{0}{1}'.format(routes[route]['domain'], expanded)
-
-    raise RouteNotFoundException('No route for resource_type: "{0}"'.format(resource_type))
+class RouteManager(object):
```

Comment:
Let me know once you get this issue resolved to move ahead with this branch https://github.com/ZeroCater/mp-slots-and-orders/pull/107